internal/provider/framework/diagnostics.go

```diff
@@ -0,0 +1,19 @@
+package framework
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+// JoinDiagnostics merges multiple sets of diag.Diagnostics.
+func JoinDiagnostics(dd ...diag.Diagnostics) diag.Diagnostics {
```

Comment:
any benefit to using this over using the `diags` package?

Reply:
Basically only readability / convenience, since often we do multiple actions which produce diagnostics but only care if there is an error at certain points, and `Diagnostics` themselves are arrays, so appending the result of each action must be done individually, which I was finding fairly noisy and cumbersome, eg: 

```go
diags.Append(req.Plan.GetAttribute(ctx, path.Root("inventory"), &data.Inventory)...)
diags.Append(req.Plan.GetAttribute(ctx, path.Root("name"), &data.Name)...)
diags.Append(req.Plan.GetAttribute(ctx, path.Root("id"), &data.Id)...)
```

vs

```go
diags.Append(framework.JoinDiagnostics(
  req.Plan.GetAttribute(ctx, path.Root("inventory"), &data.Inventory),
  req.Plan.GetAttribute(ctx, path.Root("name"), &data.Name),
  req.Plan.GetAttribute(ctx, path.Root("id"), &data.Id),
)...)
```

Which felt easier to quickly parse as "this is a set of actions we want to do and error check as a whole".

Specifically in situations where we are calling internal funcs that return diagnostics instead of having direct access to the response, it also makes some return flows a little more elegant. Since it is a fairly minor and definitely subjective improvement I'd be happy to back it out if you prefer.